doc/developer/platform/ux.md

```diff
@@ -1,20 +1,945 @@
+```
+
+But since UUIDs are just 16 bytes each, we can pack them into a much shorter
+base64-encoded string:
+
+```py
+>>> base64.urlsafe_b64encode(client_id.bytes + secret_key.bytes).decode().rstrip('=')
+iTZ3wqdMTsuATTt_oMYf3GgQTaI0F0QuphTweNrEcL4
+```
+
+When encoded in this form, we call the value an app-specific password rather
+than an API key.
+
+In the web UI, we will need to build a custom interface for managing personal
+API tokens that shows the app-specific password form alongside the client ID
+and secret.
+
+### Billing
+
+Materialize Platform will bill monthly based on usage. Accounts will be billed
+in terms of compute units, storage units, and network transfer units. A strawman
+proposal: compute units are virtual machine-hours, storage units are
+gigabyte-months, and network transfer units are gigabytes. Only outbound
+network traffic is billed.
+
+We expect metering the compute usage of clusters, the storage usage of sources
+and tables, and outbound network traffic generated by sinks and query responses,
+to be straightforward.
+
+What is not clear is how to bill for the overhead of other system components,
+like the compute usage of source ingestion. One candidate proposal is to
+additionally bill per gigabyte ingested to cover the compute cost of the
+ingestion.
+
+### Region
+
+When an account is created, it does not have any regions. The user's first step
+after registering for a new account is to choose one region to activate.
+
+Additional regions can be enabled via a "+ Enable new region" button in the
+region selector or somesuch in the web UI, or via the [global API](#global-api).
+
+To prevent disaster, regions cannot be deactivated. Users who wish to deactivate
+a region can contact support.
+
+The initially supported regions will be the same regions as supported today
+by Materialize Cloud:
+
+  * aws:us-east-1
+  * aws:eu-west-1
+
+### Cluster
+
+Clusters are created within a region via a new `CREATE CLUSTER` SQL statement:
+
+```sql
+CREATE CLUSTER <name> SIZE {xs|s|m|l|xl} [, REPLICATION FACTOR <n>];
+```
+
+Creating a cluster allocates persistent compute resources (e.g., EC2 instances)
+and begins billing the account for those resources. The SQL statement returns
+successfully as soon as the request is received.
+
+A cluster may be horizontally scaled across multiple machines. The scaling is
+deliberately hidden from users behind the opaque notion of "size".
+
+Specifying a replication factor greater than one will run multiple identical
+replicas of the cluster for redundancy. Any requests to the cluster will be
+transparently routed to any live replica, so the cluster can tolerate failures
+of `n - 1` replicas before it becomes unavailable.
+
+All machines in a replica run in the same availability zone (AZ). If a cluster
+is configured with multiple replicas, replicas are distributed across AZs as
+evenly as possible to ensure availability in the event of an AZ outage.
```

Comment:
trying to make sure I have the right picture in my head here -- a cluster can be `n` instances, all within one AZ. Each replica would be an equivalently scaled set of instances, all within different AZs

e.g. hypothetical cluster is scaled to 3 instances, replication factor = 2
us-east-1a: 3 instances for cluster
us-east-1b: 3 instances for replica 1
us-east-1c: 3 instances of replica 2

Does that seem right?

Reply:
I'm confused about the "us-east-1a" line in your example! Here are some examples that I posit are correct:

e.g. hypothetical cluster is scaled to 3 instances, replication factor = 2
us-east-1b: 3 instances for replica 0
us-east-1c: 3 instances for replica 1

e.g. hypothetical cluster is scaled to 3 instances, replication factor = 3
us-east-1a: 3 instances for replica 0
us-east-1b: 3 instances for replica 1
us-east-1c: 3 instances for replica 2

e.g. hypothetical cluster is scaled to 1 instances, replication factor = 3
us-east-1a: 1 instance for replica 0
us-east-1b: 1 instance for replica 1
us-east-1c: 1 instance for replica 2

Reply:
Ah that makes more sense, I had a primary/replica setup in mind rather than something where they're all just replicas 👍